File: scripts/python/async_version/time_checker.py
import asyncio
import ssl
import time

import aiohttp
import certifi


async def fetch(session, url):
    async with session.get(url) as response:
        return await response.text()


async def main():
    url = "https://habr.com"
    num_requests = 100

    start_time = time.time()

    ssl_context = ssl.create_default_context(cafile=certifi.where())
    async with aiohttp.ClientSession(
        connector=aiohttp.TCPConnector(ssl=ssl_context)
    ) as session:
        tasks = [fetch(session, url) for _ in range(num_requests)]
        await asyncio.gather(*tasks)

    end_time = time.time()
    execution_time = end_time - start_time

    print(f"Время выполнения: {execution_time} секунд")


if __name__ == "__main__":
    asyncio.run(main())

File: scripts/javascript/async_version/time_checker.js
const axios = require('axios');

async function fetchData() {
    const numRequests = 100;
    const startTime = Date.now();

    const requests = [];
    for (let i = 0; i < numRequests; i++) {
        const request = axios.get('https://habr.com').catch(error => {
            return null;
        });
        requests.push(request);
    }

    await Promise.all(requests);

    const endTime = Date.now();
    const executionTime = (endTime - startTime) / 1000;

    console.log(`Время выполнения: ${executionTime} секунд`);
}

fetchData();
